pkg/resource/replication_group/sdk.go

```diff
@@ -1381,29 +1381,28 @@ func (rm *resourceManager) newUpdateRequestPayload(
 func (rm *resourceManager) sdkDelete(
 	ctx context.Context,
 	r *resource,
-) (err error) {
+) (latest *resource, err error) {
 	rlog := ackrtlog.FromContext(ctx)
 	exit := rlog.Trace("rm.sdkDelete")
 	defer exit(err)
+	// if resource is already deleting, return requeue error; otherwise, initiate deletion
 	if isDeleting(r) {
-		return requeueWaitWhileDeleting
+		return r, requeueWaitWhileDeleting
 	}
 
 	input, err := rm.newDeleteRequestPayload(r)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	_, err = rm.sdkapi.DeleteReplicationGroupWithContext(ctx, input)
+	var resp *svcsdk.DeleteReplicationGroupOutput
+	_ = resp
+	resp, err = rm.sdkapi.DeleteReplicationGroupWithContext(ctx, input)
 	rm.metrics.RecordAPICall("DELETE", "DeleteReplicationGroup", err)
+	// delete call successful
 	if err == nil {
-		if foundResource, err := rm.sdkFind(ctx, r); err != ackerr.NotFound {
-			if isDeleting(foundResource) {
-				return requeueWaitWhileDeleting
-			}
-			return err
-		}
+		return r, requeueWaitWhileDeleting
```

Comment:
This logic does not take values from `resp.ReplicationGroup`.
Please add a comment here about this behavior in the code.

It is expected to work fine, as the next reconciler run (after requeue wait interval) will fetch latest state of it and lines 1389-1390 above will ensure that the correct status is updated inside reconciler->patchResource.